### DISCUSSION TITLE
Request entity too large when uploading a 11mb edge function

### DISCUSSION THREAD

**Discussion by nico1996it** [supabase/supabase#20079] [Questions]
This function run flawless when i try it locally, however when i try do deploy it: 
`functions deploy send-invoice --no-verify-jwt        
Bundling send-invoice
Deploying send-invoice (script size: 11.09MB)
Failed to create a new Function on the Supabase project: {"message":"request entity too large"}`

Is this limit real? Is it documented somewhere?

That's the full log if I deploy in debug mode 

`2023/12/31 14:16:56 Sent Header: Host [api.moby.localhost]
2023/12/31 14:16:56 Sent Header: User-Agent [Docker-Client/unknown-version (darwin)]
2023/12/31 14:16:56 Send Done
2023/12/31 14:16:56 Recv First Byte
Deploying send-invoice (script size: 11.09MB)
2023/12/31 14:16:58 DNS Start: {Host:api.supabase.com}
2023/12/31 14:16:58 DNS Done: {Addrs:[{IP:104.21.30.61 Zone:} {IP:172.67.172.49 Zone:} {IP:2606:4700:3035::6815:1e3d Zone:} {IP:2606:4700:3035::ac43:ac31 Zone:}] Err:<nil> Coalesced:false}
2023/12/31 14:16:58 Connect Start: tcp 104.21.30.61:443
2023/12/31 14:16:58 Connect Done: tcp 104.21.30.61:443
2023/12/31 14:16:58 TLS Start
2023/12/31 14:16:59 TLS Done: {Version:772 HandshakeComplete:true DidResume:false CipherSuite:4865 NegotiatedProtocol:h2 NegotiatedProtocolIsMutual:true ServerName:api.supabase.com PeerCertificates:[0x140002e2580 0x140002e2b00 0x140002e3080 0x140002e3600] VerifiedChains:[[0x140002e3b80 0x140002e4100 0x140002e4680]] SignedCertificateTimestamps:[] OCSPResponse:[48 130 1 84 10 1 0 160 130 1 77 48 130 1 73 6 9 43 6 1 5 5 7 48 1 1 4 130 1 58 48 130 1 54 48 129 190 161 52 48 50 49 11 48 9 6 3 85 4 6 19 2 85 83 49 22 48 20 6 3 85 4 10 19 13 76 101 116 39 115 32 69 110 99 114 121 112 116 49 11 48 9 6 3 85 4 3 19 2 69 49 24 15 50 48 50 51 49 50 50 57 48 52 51 54 48 48 90 48 117 48 115 48 75 48 9 6 5 43 14 3 2 26 5 0 4 20 239 144 1 112 221 88 143 42 101 30 33 81 31 220 208 187 111 81 43 171 4 20 90 243 237 43 252 54 194 55 121 185 82 48 234 84 111 207 85 203 46 172 2 18 4 196 117 114 101 160 49 96 181 199 143 221 225 66 79 68 214 24 128 0 24 15 50 48 50 51 49 50 50 57 48 52 51 54 48 48 90 160 17 24 15 50 48 50 52 48 49 48 53 48 52 51 53 53 56 90 48 10 6 8 42 134 72 206 61 4 3 3 3 103 0 48 100 2 48 10 9 184 215 139 23 96 253 223 128 80 232 110 225 89 249 142 38 46 218 113 165 205 176 46 26 229 189 53 119 154 92 20 95 252 115 35 222 198 121 108 60 218 131 64 109 7 128 2 48 124 28 71 34 30 79 150 185 205 6 197 35 193 251 12 120 9 209 143 194 13 117 158 214 197 85 196 232 133 60 208 207 95 146 144 97 52 151 29 163 11 204 173 226 11 243 203 38] TLSUnique:[] ekm:0x105206240}
2023/12/31 14:16:59 Sent Header: :authority [api.supabase.com]
2023/12/31 14:16:59 Sent Header: :method [GET]
2023/12/31 14:16:59 Sent Header: :path [/v1/projects/ijijjcrvrrklzqylaybb/functions/send-invoice]
2023/12/31 14:16:59 Sent Header: :scheme [https]
2023/12/31 14:16:59 Sent Header: authorization [Bearer sbp_b41148559e2209d010a83b6218a4da5474300c31]
2023/12/31 14:16:59 Sent Header: user-agent [SupabaseCLI/1.123.4]
2023/12/31 14:16:59 Sent Header: accept-encoding [gzip]
2023/12/31 14:16:59 Send Done
2023/12/31 14:16:59 Recv First Byte
2023/12/31 14:16:59 Sent Header: :authority [api.supabase.com]
2023/12/31 14:16:59 Sent Header: :method [POST]
2023/12/31 14:16:59 Sent Header: :path [/v1/projects/ijijjcrvrrklzqylaybb/functions?entrypoint_path=file%3A%2F%2F%2Fhome%2Fdeno%2Ffunctions%2Fsend-invoice%2Findex.ts&import_map_path=file%3A%2F%2F%2Fhome%2Fdeno%2Fimport_map.json&name=send-invoice&slug=send-invoice&verify_jwt=false]
2023/12/31 14:16:59 Sent Header: :scheme [https]
2023/12/31 14:16:59 Sent Header: content-type [application/vnd.denoland.eszip]
2023/12/31 14:16:59 Sent Header: authorization [Bearer sbp_b41148559e2209d010a83b6218a4da5474300c31]
2023/12/31 14:16:59 Sent Header: user-agent [SupabaseCLI/1.123.4]
2023/12/31 14:16:59 Sent Header: content-length [11090347]
2023/12/31 14:16:59 Sent Header: accept-encoding [gzip]
2023/12/31 14:17:04 Send Done
2023/12/31 14:17:08 Recv First Byte
Failed to create a new Function on the Supabase project: {"message":"request entity too large"}`

**Comment by GaryAustin1** [comment]
![image](https://github.com/supabase/supabase/assets/54564956/1d308ebc-f5c6-42b6-ba46-3d78882e2034)

**Reply by martinrobaey5th** [reply]
Dear,

Would you have an option to reduce the scrip size ?

**Reply by GaryAustin1** [reply]
I believe you control your code and the modules you load.

**Reply by martinrobaey5th** [reply]
But I cannot reduce the size of the module I import, do you know a way to maybe minify the size of the bundle being send to supabase when deploying the function ?

**Reply by GaryAustin1** [reply]
This is the only guide I can find on it. https://supabase.com/docs/guides/functions/debugging#finding-bundle-size
